explorer/src/pages/GatewayDetail/index.tsx

```diff
@@ -0,0 +1,182 @@
+import * as React from 'react';
+import { Alert, AlertTitle, Box, CircularProgress, Grid } from '@mui/material';
+import { useParams } from 'react-router-dom';
+import { GatewayResponseItem } from '../../typeDefs/explorer-api';
+import { ColumnsType, DetailTable } from '../../components/DetailTable';
+import { gatewayEnrichedToGridRow, GatewayEnrichedRowType } from '../../components/Gateways';
+import { ComponentError } from '../../components/ComponentError';
+import { ContentCard } from '../../components/ContentCard';
+import { TwoColSmallTable } from '../../components/TwoColSmallTable';
+import { UptimeChart } from '../../components/UptimeChart';
+// import { GatewayDetailSection } from '../../components/Gateways/DetailSection';
+import { GatewayContextProvider, useGatewayContext } from '../../context/gateway';
+import { useMainContext } from '../../context/main';
+import { Title } from '../../components/Title';
+
+const columns: ColumnsType[] = [
+  {
+    field: 'identity_key',
+    title: 'Identity Key',
+    headerAlign: 'left',
+    width: 230,
+  },
+  {
+    field: 'bond',
+    title: 'Bond',
+    flex: 1,
+    headerAlign: 'left',
+  },
+  {
+    field: 'routing_score',
+    title: 'Routing Score',
+    flex: 1,
+    headerAlign: 'left',
+    tooltipInfo:
+      'Routing score is relative to that of the network. Each time a gateway is tested, the test packets have to go through the full path of the network (gateway + 3 nodes). If a node in the path drop packets it will affect the score of the gateway and other nodes in the test.',
+  },
+  {
+    field: 'avg_uptime',
+    title: 'Avg. Score',
+    flex: 1,
+    headerAlign: 'left',
+    tooltipInfo: 'Is the average routing score in the last 24 hours',
+  },
+  {
+    field: 'host',
+    title: 'IP',
+    headerAlign: 'left',
+    width: 99,
+  },
+  {
+    field: 'location',
+    title: 'Location',
+    headerAlign: 'left',
+    flex: 1,
+  },
+  {
+    field: 'owner',
+    title: 'Owner',
+    headerAlign: 'left',
+    flex: 1,
+  },
+];
+
+/**
+ * Shows gateway details
+ */
+const PageGatewayDetailsWithState: React.FC<{ selectedGateway: GatewayResponseItem | undefined }> = ({
+  selectedGateway,
+}) => {
+  const [enrichGateway, setEnrichGateway] = React.useState<GatewayEnrichedRowType>();
+  const [status, setStatus] = React.useState<number[] | undefined>();
+  const { uptimeReport, uptimeStory } = useGatewayContext();
+
+  React.useEffect(() => {
+    if (uptimeReport?.data && selectedGateway) {
+      setEnrichGateway(gatewayEnrichedToGridRow(selectedGateway, uptimeReport.data));
+    }
+  }, [uptimeReport, selectedGateway]);
+
+  React.useEffect(() => {
+    if (enrichGateway) {
+      setStatus([enrichGateway?.mix_port, enrichGateway?.clients_port]);
```

Comment:
shouldnt need the optional chaining operator (`?`) here as we have determined that `enrichGateway` exists through if statement